src/pkg/caendr/caendr/services/cloud/google_analytics_4.py

```diff
@@ -0,0 +1,78 @@
+import os
+import pandas as pd
+
+from datetime import datetime
+from oauth2client.service_account import ServiceAccountCredentials
+
+from caendr.services.cloud.secret import get_secret
+from caendr.services.cloud.service_account import get_service_account_credentials
+from caendr.services.cloud.storage import get_blob, download_blob_as_dataframe
+from caendr.utils.env import get_env_var
+
+from .discovery import use_service
+
+
+GOOGLE_ANALYTICS_SERVICE_ACCOUNT_NAME = get_env_var('GOOGLE_ANALYTICS_SERVICE_ACCOUNT_NAME')
+
+
+def get_analytics_credentials():
+  '''
+    Helper function to get the service account credentials that authorize access to Google Analytics
+  '''
+  service_credentials = get_service_account_credentials(get_secret(GOOGLE_ANALYTICS_SERVICE_ACCOUNT_NAME))
+  scope = ['https://www.googleapis.com/auth/analytics.readonly']
+  return ServiceAccountCredentials.from_json_keyfile_dict(service_credentials, scope)
+
+
+@use_service('analyticsdata', 'v1beta', credentials=get_analytics_credentials)
+def get_weekly_visits_ga4(SERVICE):
+  """ Get reports from Google Analytics 4 """
+
+  property_id = '361534565'
+  property = f"properties/{property_id}"
+  request = {
+    "dateRanges": [
+    {
+        "startDate": "2023-04-01",
```

Comment:
consider creating a constant here with the `startDate`, eg: GOOGLE_ANALYTICS_V4_START_DATE = "2023-04-01"`.

Reply:
fixed in the commit [ac2f630](https://github.com/AndersenLab/CAENDR/pull/408/commits/ac2f6309f446117b209c29df9d25ebfc661f045e)